ouroboros-consensus/src/Ouroboros/Storage/FS/Sim/FsTree.hs

```diff
@@ -189,8 +193,8 @@ getDir fp = getConst . alterDir fp (Const . Left) errNotExist (Const . Right)
 -------------------------------------------------------------------------------}
 
 -- | Make sure the specified file exists; create it if necessary.
-touch :: Monoid a => FsPath -> FsTree a -> Either FsTreeError (FsTree a)
-touch fp = alterFile fp Left (Right mempty) Right
+touch :: Monoid a => FsPath -> Bool -> FsTree a -> Either FsTreeError (FsTree a)
+touch fp excl = alterFile fp Left (Right mempty) (\a -> if excl then Left (FsExists fp) else Right a)
```

Comment:
Let's document that we _need_ this to work with both inclusive and exclusive mode because of how we implement open in the model. Might want to rename this function.